From dd77826420f2c4bb53dbb310f8b22e2c67d05b17 Mon Sep 17 00:00:00 2001
From: Alexandre Jutras <alexandre.jutras@linaro.org>
Date: Fri, 14 Feb 2020 09:21:35 -0500
Subject: [PATCH 3/6] vpu: Request HEVC in bytes-stream format for SDP

Signed-off-by: Alexandre Jutras <alexandre.jutras@linaro.org>
---
 plugins/vpu/gstvpu.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/plugins/vpu/gstvpu.h b/plugins/vpu/gstvpu.h
index 461e4d6..8d0788e 100755
--- a/plugins/vpu/gstvpu.h
+++ b/plugins/vpu/gstvpu.h
@@ -41,7 +41,11 @@ typedef struct
 } VPUMapper;
 
 static VPUMapper vpu_mappers[] = {
+#if ENABLE_SDP
+  {VPU_V_HEVC, "video/x-h265, stream-format=(string)byte-stream"},
+#else
   {VPU_V_HEVC, "video/x-h265"},
+#endif
   {VPU_V_VP9, "video/x-vp9"},
   {VPU_V_VP8, "video/x-vp8"},
   {VPU_V_VP6, "video/x-vp6-flash"},
-- 
2.17.1

