From cad6a60cf1c8442752e20e069f82a437d79bb07f Mon Sep 17 00:00:00 2001
From: Alexandre Jutras <alexandre.jutras@linaro.org>
Date: Thu, 28 Nov 2019 13:20:40 -0500
Subject: [PATCH] aacparse: [HACK] no codec_data issue

Signed-off-by: Alexandre Jutras <alexandre.jutras@linaro.org>
---
 gst/audioparsers/gstaacparse.c | 25 +++++++++++++++++++++++++
 1 file changed, 25 insertions(+)

diff --git a/gst/audioparsers/gstaacparse.c b/gst/audioparsers/gstaacparse.c
index f7a0cc4ac..876b08b4d 100644
--- a/gst/audioparsers/gstaacparse.c
+++ b/gst/audioparsers/gstaacparse.c
@@ -384,7 +384,32 @@ gst_aac_parse_sink_setcaps (GstBaseParse * parse, GstCaps * caps)
 
     if (g_strcmp0 (stream_format, "raw") == 0) {
       GST_ERROR_OBJECT (parse, "Need codec_data for raw AAC");
+
+#if 0
       return FALSE;
+#else
+      aacparse->header_type = DSPAAC_HEADER_NONE;
+      aacparse->mpegversion = 4;
+      aacparse->object_type = 2;
+      aacparse->sample_rate = 44100;
+      aacparse->channels = 2;
+      aacparse->frame_samples = 1024;
+
+     GST_DEBUG ("No codec_data for raw format: Assuming object_type=%d, sample_rate=%d, channels=%d, "
+          "samples=%d", aacparse->object_type, aacparse->sample_rate,
+          aacparse->channels, aacparse->frame_samples);
+
+      /* arrange for metadata and get out of the way */
+      gst_aac_parse_set_src_caps (aacparse, caps);
+      if (aacparse->header_type == aacparse->output_header_type)
+        gst_base_parse_set_passthrough (parse, TRUE);
+
+      /* input is already correctly framed */
+      gst_base_parse_set_min_frame_size (parse, RAW_MAX_SIZE);
+
+      return TRUE;
+#endif
+
     } else {
       aacparse->sample_rate = 0;
       aacparse->channels = 0;
-- 
2.17.1

