From 2c0daf5ef39bac1cfee4a69a404be45105b55b82 Mon Sep 17 00:00:00 2001
From: Peter Griffin <peter.griffin@linaro.org>
Date: Thu, 6 Aug 2020 15:16:51 +0100
Subject: [PATCH] backend-drm/drm.c: Use DRM_FORMAT_P010 with v4.14 kernel

Signed-off-by: Peter Griffin <peter.griffin@linaro.org>
---
 libweston/backend-drm/drm.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libweston/backend-drm/drm.c b/libweston/backend-drm/drm.c
index 2323e93b..3a729915 100644
--- a/libweston/backend-drm/drm.c
+++ b/libweston/backend-drm/drm.c
@@ -3017,7 +3017,7 @@ drm_import_dmabuf(struct weston_compositor *compositor,
 
 		for (i = 0; i < p->count_formats; i++) {
 			if (p->formats[i].format == dmabuf->attributes.format
-				&& dmabuf->attributes.format == DRM_FORMAT_NV12_10LE40)
+				&& dmabuf->attributes.format == DRM_FORMAT_P010)
 				return true;
 		}
 	}
-- 
2.17.1

